pkg/util/appgroup.go

```diff
@@ -0,0 +1,316 @@
+/*
```

Comment:
All this code is currently coupled with `pkg/controller/appgroup.go`. What's the likelihood the code gets reused some place other than `pkg/controller/appgroup.go`?

Also, worth introducing `appgroup_test.go` testing individual sorting algorithm. `TestAppGroupController_Run` is testing all the bits from the top level making it harder to test specific use cases for e.g. the sorting algorithms.